### PR TITLE
Add comments

### PR DESCRIPTION
**Pull request by LianeWalter**
## What was done?
Comments can now be added to (existing) posts. 
Comments for a specific post can be retrieved with the post-id.
The database structure for comments consists of an entity table and a closure table to support hierarchical comments in the future :) 

The test for the PostController was failing due to a missing dependency of the PostRepository, this was fixed (albeit a little lazily).

## What was not done?
Adding comments to comments was not part of this issue.

### CODE REVIEW

File: web-server/src/comment/comment.controller.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            new ValidationPipe({
+                whitelist: true,
+                transform: true,
+            }),
+        )
+        data: CommentDto,
+    ): Promise<Comment> {
+        const post: Post = await this.postRepository.findOne(data.postId);
+
+        if (post === undefined) {
```
Comment (tetrohed) [inline web-server/src/comment/comment.controller.ts:37]: we might want to consider using a not found exception instead of bad request, however Im always unsure which one is the better practice in case of posts that reference other ids => maybe you could find a reference to standard practice?

File: web-server/src/comment/comment.controller.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            throw new HttpException('Incorrect postId submitted.', HttpStatus.BAD_REQUEST);
```
Comment (tetrohed) [inline web-server/src/comment/comment.controller.ts:38]: also how can we unit test this?

File: web-server/src/comment/comment.entity.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    Entity,
+    PrimaryGeneratedColumn,
+    Tree,
+    ManyToOne,
+    TreeChildren,
+    TreeParent
+} from 'typeorm';
+
+@Entity()
+@Tree("closure-table")
```
Comment (tetrohed) [inline web-server/src/comment/comment.entity.ts:14]: nice choice, that's exactly what I had in mind

File: web-server/src/comment/comment.entity.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+export class Comment {
+    @PrimaryGeneratedColumn()
+    readonly id: number;
+
+    @CreateDateColumn()
```
Comment (tetrohed) [inline web-server/src/comment/comment.entity.ts:19]: date column might be a trouble in tests(since it uses the dates from database implementations) -> any suggestions as to how we can improve it?

File: web-server/src/comment/comment.controller.ts (hunk 1 above)
Comment (LianeWalter) [reply web-server/src/comment/comment.controller.ts:37]: You're right 400 is more about syntax. I guess 422 is a better status code as it is about semantics:  https://tools.ietf.org/html/rfc4918#section-11.2 

The Adyen API uses that response code for example.

File: web-server/src/comment/comment.entity.ts (hunk 4 above)
Comment (LianeWalter) [reply web-server/src/comment/comment.entity.ts:19]: We could generate the date in the backend and then save it along with the other data.

File: web-server/src/comment/comment.entity.ts (hunk 3 above)
Comment (LianeWalter) [reply web-server/src/comment/comment.entity.ts:14]: 😄 👍

File: web-server/src/comment/comment.controller.ts (hunk 2 above)
Comment (LianeWalter) [reply web-server/src/comment/comment.controller.ts:38]: With the current approach, I'd have to mock the repo call.
In reality, The repo call would be in a service and I would unit test that functionality there. 
In true YAGNI fashion, I decided against adding that much boilerplate code for just 4 lines of code, so no service. 

I'll add a simple unit test.

Comment (tetrohed) [reply web-server/src/comment/comment.controller.ts]: sounds good.

For future reference, exactly because of this conundrum I ended up spinning up a mini database per testcase and doing api testing for happy paths.

here is gist straight out of our mono repo

https://gist.github.com/tetrohed/5e9a3c122007791e588c211448453ec1

for the `services/edge cases/fail paths` I still do unit tests.

File: web-server/src/comment/comment.entity.ts (hunk 4 above)
Comment (tetrohed) [reply web-server/src/comment/comment.entity.ts:19]: exactly what we are doing in our mono repo -> we made a date service that can be mocked out in testcases.

File: web-server/src/comment/comment.controller.ts (hunk 1 above)
Comment (tetrohed) [reply web-server/src/comment/comment.controller.ts:37]: great
